activerecord/test/cases/relations_test.rb

```diff
@@ -1043,7 +1043,9 @@ def test_any
     posts.where(id: nil).any?
 
     assert_queries(3) do
-      assert posts.any? # Uses COUNT()
+      assert_sql(/SELECT\ +1/) do
```

Comment:
It wasn't immediately clear what this SQL assertions is testing for. I would consider using something like `/SELECT  1 AS one/` instead.

Reply:
Because for some reason
```sql
SELECT  1 AS one
```
is generated instead of
```sql
SELECT 1 AS one
```
I wasn't sure why is this extra space here. This is a quirk of `connection.select_values` so I thought it'd be better not to conform to it.

Reply:
I simplified all the regexes with extra-space ones.

Reply:
For some reason, Rails adds extra space after `SELECT` if there's `LIMIT`:
```ruby
[17] pry(main)> puts Post.all.select(1).to_sql
SELECT 1 FROM "posts"
=> nil
[18] pry(main)> puts Post.all.select(1).limit(1).to_sql
SELECT  1 FROM "posts" LIMIT 1
```

Reply:
Let stick with 
```ruby
assert_sql(/SELECT  1 AS one/) do
...
end
```
until the Arel is fixed.

Reply:
I tried to fix the double-space issue in #33977

Reply:
Nice work! Rebase and I think this can be merged.

Reply:
#33977 was merged so I finally can rewrite these tests with single space :)